<?xml version="1.0"?>
<!-- Robot Name -->
<robot name="delta_robot">

    <!-- Base -->
    <link name="base_link">
        <visual>
            <geometry>
                <cylinder length="0.02" radius="0.180"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0"/>
        </visual>
    </link>

<!-- Arm 1 -->
 <link name="arm_1">
 <visual>
 <geometry>
 <box size="0.300 0.02 0.02"/>
 </geometry>
 <origin rpy="0 0 0" xyz="0.150 0 0"/>
 <material name="blue">
 <color rgba="0 0 1 1"/>
 </material>
 </visual>
 </link>
 <joint name="base_to_arm_1" type="revolute">
 <axis xyz="0 1 0"/>
 <limit effort="1000.0" lower="-1.5708" upper="1.5708" velocity="0.5"/>
 <parent link="base_link"/>
 <child link="arm_1"/>
 <origin rpy="0 0 -1.5708" xyz="0 -0.180 0"/>
 </joint>

<!-- Down Part Arm 1 -->
<link name="down_arm_1">
 <visual>
 <geometry>
 <box size="0.800 0.02 0.02"/>
 </geometry>
 <origin rpy="0 0 0" xyz="0.400 0 0"/>
 <material name="blue">
 <color rgba="0 0 0.8 1"/>
 </material>
 </visual>
 </link>
 <joint name="down_arm_1_to_arm_1" type="revolute">
 <axis xyz="0 1 0"/>
 <limit effort="1000.0" lower="-1.5708" upper="1.5708" velocity="0.5"/>
 <parent link="arm_1"/>
 <child link="down_arm_1"/>
 <origin rpy="0 1.5708 0" xyz="0.300 0 0"/>
 </joint>
<!-- End efector -->
 <link name="tool0">
 <visual>
 <geometry>
 <cylinder length="0.02" radius="0.100"/>
 </geometry>
 <origin rpy="0 0 0" xyz="0.100 0 0"/>
 <material name="blue">
 <color rgba="0 0 0.8 1"/>
 </material>
 </visual>
 </link>
 <joint name="tool0_to_down_arm_1" type="revolute">
 <axis xyz="0 1 0"/>
 <limit effort="1000.0" lower="-1.5708" upper="1.5708" velocity="0.5"/>
 <parent link="down_arm_1"/>
 <child link="tool0"/>
 <origin rpy="0 1.5708 0" xyz="0.800 0 0"/>
 </joint>
<!-- Arm 2 -->
 <link name="arm_2">
 <visual>
 <geometry>
 <box size="0.300 0.02 0.02"/>
 </geometry>
 <origin rpy="0 0 0" xyz="0.150 0 0"/>
 <material name="blue">
 <color rgba="0 0 1 1"/>
 </material>
 </visual>
 </link>
 <joint name="base_to_arm_2" type="revolute">
 <axis xyz="0 1 0"/>
 <limit effort="1000.0" lower="-1.5708" upper="1.5708" velocity="0.5"/>
 <parent link="base_link"/>
 <child link="arm_2"/>
 <origin rpy="0 0 0.5236" xyz="0.155885 0.09 0"/>
 </joint>
<!-- Down Part Arm 2 -->
 <link name="down_arm_2">
 <visual>
 <geometry>
 <box size="0.800 0.02 0.02"/>
 </geometry>
 <origin rpy="0 0 0" xyz="0.400 0 0"/>
 <material name="blue">
 <color rgba="0 0 0.8 1"/>
 </material>
 </visual>
 </link>
 <joint name="down_arm_2_to_arm_2" type="revolute">
 <axis xyz="0 1 0"/>
 <limit effort="1000.0" lower="-1.5708" upper="1.5708" velocity="0.5"/>
 <parent link="arm_2"/>
 <child link="down_arm_2"/>
 <origin rpy="0 1.5708 0" xyz="0.300 0 0"/>
 </joint>
<!-- Arm 3 -->
 <link name="arm_3">
 <visual>
 <geometry>
 <box size="0.300 0.02 0.02"/>
 </geometry>
 <origin rpy="0 0 0" xyz="0.150 0 0"/>
 <material name="blue">
 <color rgba="0 0 1 1"/>
 </material>
 </visual>
 </link>
 <joint name="base_to_arm_3" type="revolute">
 <axis xyz="0 1 0"/>
 <limit effort="1000.0" lower="-1.5708" upper="1.5708" velocity="0.5"/>
 <parent link="base_link"/>
 <child link="arm_3"/>
 <origin rpy="0 0 2.618" xyz="-0.155885 0.09 0"/>
 </joint>
<!-- Down Part Arm 3 -->
 <link name="down_arm_3">
 <visual>
 <geometry>
 <box size="0.800 0.02 0.02"/>
 </geometry>
 <origin rpy="0 0 0" xyz="0.400 0 0"/>
 <material name="blue">
 <color rgba="0 0 0.8 1"/>
 </material>
 </visual>
  </link>
 <joint name="down_arm_3_to_arm_3" type="revolute">
 <axis xyz="0 1 0"/>
 <limit effort="1000.0" lower="-1.5708" upper="1.5708" velocity="0.5"/>
 <parent link="arm_3"/>
 <child link="down_arm_3"/>
 <origin rpy="0 1.5708 0" xyz="0.300 0 0"/>
 </joint>
</robot>
